orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/ContextParameterProcessor.groovy

```diff
@@ -177,8 +184,8 @@ class ContextParameterProcessor {
     }
   }
 
-  static class WhitelistTypeLocator implements TypeLocator {
-    private final Set<Class<?>> allowedTypes = Collections.unmodifiableSet([
+  static class TypeWhitelist {
```

Comment:
now differentiated between what can be instantiated (allowedTypes) and what return types are allowed (allowedReturnTypes)